src/pages/index.js

```diff
@@ -92,38 +124,106 @@ function getCardElement(data) {
 }
 
-// Handle profile edit form submission
+function handleDeleteSubmit() {
+  if (selectedCard && selectedCardId) {
+    const submitButton = deleteBtn;
+    submitButton.textContent = "Deleting...";
+    submitButton.disabled = true;
+
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        submitButton.textContent = "Delete";
+        submitButton.disabled = false;
+        selectedCard = null;
+        selectedCardId = null;
+      });
+  }
+}
+
+deleteBtn.addEventListener("click", handleDeleteSubmit);
+
 function handleEditFormSubmit(e) {
   e.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitButton = e.submitter;
+  submitButton.textContent = "Saving...";
+  submitButton.disabled = true;
+
+  api.editUserInfo({ 
+    name: editModalNameInput.value, 
+    about: editModalDescriptionInput.value 
+  })
+  .then(() => {
+    profileName.textContent = editModalNameInput.value;
+    profileDescription.textContent = editModalDescriptionInput.value;
+    closeModal(editModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+    submitButton.textContent = "Save";
+    submitButton.disabled = false;
+  });
 }
 
-// Handle add card form submission
 function handleAddCardSubmit(e) {
   e.preventDefault();
+  const submitButton = e.submitter;
+  submitButton.textContent = "Saving...";
+  submitButton.disabled = true;
+
+  api.addPostCards(cardNameInput.value, cardLinkInput.value)
+    .then((card) => {
+      const newCard = getCardElement({
+        name: card.name,
+        link: card.link,
+        _id: card._id,
+        isLiked: card.isLiked
+      });
+      cardsList.prepend(newCard);
+      cardForm.reset();
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      submitButton.disabled = false;
+      disabledButton(cardSubmitBtn, settings);
+    });
+}
 
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-
-  // Reset the form and disable the button after submission
-  cardForm.reset();
-  disabledButton(cardSubmitBtn, settings);
-
-  closeModal(cardModal);
+function handleAvatarSubmit(e) {
+  e.preventDefault();
+  const submitButton = e.submitter;
+  submitButton.textContent = "Saving...";
+  submitButton.disabled = true;
+
+  api.editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileImage.src = data.avatar;
+      avatarForm.reset();
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      submitButton.disabled = false;
+      disabledButton(avatarSubmitBtn, settings);
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.


See my remarks for `handleAddCardSubmit`, how to make it correctly